test/integration/src/main/java/io/pravega/test/integration/utils/TestUtils.java

```diff
@@ -0,0 +1,209 @@
+     */
+    @SneakyThrows
+    public static void writeDataToStream(String scope, String stream, String message, ClientConfig writerClientConfig) {
+        writeDataToStream(scope, stream, message, 1, writerClientConfig);
+    }
+
+    /**
+     * Write the specified number of messages to the specified {@code scope}/{@code stream}, using the
+     * provided {@writerClientConfig}.
+     *
+     * @param scope the scope
+     * @param stream the stream
+     * @param message the event message to write. If it is null, a default message will be used.
+     * @param numMessages the number of event messages to write
+     * @param writerClientConfig the {@link ClientConfig} object to use to connect to the server
+     * @throws NullPointerException if {@code scope} or {@code stream} or {@writerClientConfig} is null
+     * @throws IllegalArgumentException if {@code numMessages} < 1
+     * @throws RuntimeException if any exception is thrown by the client
+     */
+    public static void writeDataToStream(@NonNull String scope, @NonNull String stream, String message, int numMessages,
+                                         @NonNull ClientConfig writerClientConfig) {
+        Preconditions.checkArgument(numMessages > 0);
+        if (message == null) {
+            message = "Test message";
+        }
+        @Cleanup final EventStreamClientFactory writerClientFactory = EventStreamClientFactory.withScope(scope,
+                writerClientConfig);
+
+        @Cleanup final EventStreamWriter<String> writer = writerClientFactory.createEventWriter(stream,
+                new JavaSerializer<String>(),
+                EventWriterConfig.builder().build());
+        for (int i = 0; i < numMessages; i++) {
+            writer.writeEvent(message);
+        }
+        writer.flush();
+        log.info("Wrote {} message(s) to the stream {}/{}", numMessages, scope, stream);
+    }
+
+    /**
+     * Returns the next unread message from the specified {@code scope}/{@code stream}.
+     *
+     * @param scope the scope
+     * @param stream the stream
+     * @param readerClientConfig the {@link ClientConfig} object to use to connect to the server
+     * @param readerGroup the name of the reader group application
+     * @return the event message
+     */
+    public static String readNextEventMessage(String scope, String stream, ClientConfig readerClientConfig,
+                                              String readerGroup) {
+        List<String> messages = readNextEventMessages(scope, stream, 1, readerClientConfig, readerGroup);
+        if (messages == null || messages.size() == 0) {
+            return null;
+        } else {
+            return messages.get(0);
+        }
+    }
+
+    /**
+     * Returns the specified number of unread messages from the given {@code scope}/{@code stream}.
+     *
+     * @param scope the scope
+     * @param stream the stream
+     * @param numMessages the number of event messages to write
```

Comment:
typo.

Reply:
Great catch - truly a hard to find one! I've fixed it now.